dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeartBeatTask.java

```diff
@@ -70,7 +71,8 @@ public void run() {
                     if (lastRead != null && now - lastRead > heartbeatTimeout) {
                         logger.warn("Close channel " + channel
                                 + ", because heartbeat read idle time out: " + heartbeatTimeout + "ms");
-                        if (channel instanceof Client) {
+
+                        if (channel instanceof Client && channel.getUrl().getParameter(Constants.HEARTBEAT_RECONNECT, Constants.DEFAULT_HEARTBEAT_RECONNECT)) {
```

Comment:
This op can not make the hanging request return directly. It just control the reconnect.
I didn't think this pr can solve this problem.
Maybe u should fix or send another pr 
:)

Reply:
Agree. This patch didn't solve the problem